prow.sh

```diff
@@ -0,0 +1,667 @@
+
+# The hostpath deployment script is searched for in several places.
+#
+# - The "deploy" directory in the current repository: this is useful
+#   for the situation that a component becomes incompatible with the
+#   shared deployment, because then it can (temporarily!) provide its
+#   own example until the shared one can be updated; it's also how
+#   csi-driver-host-path itself provides the example.
+#
+# - CSI_PROW_HOSTPATH_VERSION of the CSI_PROW_HOSTPATH_REPO is checked
+#   out: this allows other repos to reference a version of the example
+#   that is known to be compatible.
+#
+# - The csi-driver-host-path/deploy directory has multiple sub-directories,
+#   each with different deployments (stable set of images for Kubernetes 1.13,
+#   stable set of images for Kubernetes 1.14, canary for latest Kubernetes, etc.).
+#   This is necessary because there may be incompatible changes in the
+#   "API" of a component (for example, its command line options or RBAC rules)
+#   or in its support for different Kubernetes versions (CSIDriverInfo as
+#   CRD in Kubernetes 1.13 vs builtin API in Kubernetes 1.14).
+#
+#   When testing an update for a component in a PR job, the
+#   CSI_PROW_DEPLOYMENT variable can be set in the
+#   .prow.sh of each component when there are breaking changes
+#   that require using a non-default deployment. Otherwise
+#   a deployment named "kubernetes-x.yy" is used.
+#   "none" disables the deployment of the hostpath driver.
+#
+# When no deploy script is found (nothing in `deploy` directory,
+# CSI_PROW_HOSTPATH_REPO=none), nothing gets deployed.
+configvar CSI_PROW_HOSTPATH_VERSION master "hostpath driver"
+configvar CSI_PROW_HOSTPATH_REPO https://github.com/kubernetes-csi/csi-driver-host-path "hostpath repo"
+# Ignore 'See if you can use ${variable//search/replace} instead.'
+# shellcheck disable=SC2001
+configvar CSI_PROW_DEPLOYMENT "kubernetes-$(echo "${CSI_PROW_KUBERNETES_VERSION}" | sed -e 's/\([0-9]*\)\.\([0-9]*\).*/\1.\2/')" "deployment"
+
+# If CSI_PROW_HOSTPATH_CANARY is set (typically to "canary", but also
+# "1.0-canary"), then all image versions are replaced with that
+# version tag.
+configvar CSI_PROW_HOSTPATH_CANARY "" "hostpath image"
+
+# The E2E testing can come from an arbitrary repo. The expectation is that
+# the repo supports "go test ./test/e2e -args --storage.testdriver" (https://github.com/kubernetes/kubernetes/pull/72836)
+# after setting KUBECONFIG. As a special case, if the repository is Kubernetes,
+# then `make WHAT=test/e2e/e2e.test` is called first to ensure that
+# all generated files are present.
+#
+# CSI_PROW_E2E_REPO=none disables E2E testing.
+#
+# TODO (?): support also an E2E test suite in the current repository.
+# TODO: upstream Kubernetes
+configvar CSI_PROW_E2E_REPO https://github.com/pohly/kubernetes "E2E repo"
+configvar CSI_PROW_E2E_VERSION storage-external-snapshot "E2E version"
```

Comment:
what is this, the branch name?

Reply:
Branch name, revision or tag. Basically anything that git can check out.